apps/contributions/filters.py

```diff
@@ -60,13 +61,21 @@ def filter_queryset_by_rp(self, queryset, revenue_program_id: str):
             | Q(contribution_metadata__contains={"revenue_program_id": revenue_program_id})
         )
 
+    def filter_intervals(self, queryset, request):
+        if (interval := request.query_params.get("interval")) == ContributionInterval.ONE_TIME.value:
+            return queryset.filter(interval=ContributionInterval.ONE_TIME)
+        elif interval == "recurring":
+            return queryset.exclude(interval=ContributionInterval.ONE_TIME)
+        return queryset
+
```

Comment:
Minor suggestion, but this is a good place for match/case:

```suggestion
        match request.query_params.get("interval"):
            case ContributionInterval.ONE_TIME.value:
                return queyrset.filter(interval=ContributionInterval.ONE_TIME)
            case "recurring":
                return queryset.exclude(ContributionInterval.ONE_TIME)
            case _:
                return queryset
```

Reply:
Consider putting "recurring" into a constant at top of file (all caps), and then can import that var in tests.

Reply:
I tried adding the variable at the top of the file but when trying to use it inside of the match/case it didn't work.
![image](https://github.com/newsrevenuehub/rev-engine/assets/9381325/75908779-497f-43f2-8038-5eeb19b6d668)

I was able to make it work by adding it inside `PortalContributionFilter`. Is this a good workaround?
![image](https://github.com/newsrevenuehub/rev-engine/assets/9381325/2e9c8318-3f06-49ee-9b70-d1c41fae5b2d)
![image](https://github.com/newsrevenuehub/rev-engine/assets/9381325/a2dc9358-c5b5-4ed5-b145-9799b0f17aee)